elastic_service/elastic_handler.go

```diff
@@ -72,14 +74,13 @@ func (e *ElasticHandler) DeleteBook(ctx context.Context, id string) error {
 	if err != nil {
 		return errors.Wrap(err, fmt.Sprintf("couldn't delete book with id: %s", id))
 	}
-	return err
+	return nil
 }
 
-func buildSearchQuery(client *elastic.Client, title string, author string, priceRange string) (s *elastic.SearchService, err error) {
-
+func buildSearchQuery(client *elastic.Client, searchParams map[string]string) (s *elastic.SearchService, err error) {
```

Comment:
```suggestion
func buildSearchQuery(client *elastic.Client, fields map[string]string) (s *elastic.SearchService, err error) {
```

Let's return an `elastic.Query` and not `elastic.SearchService`